packages/marble-api/openapis/marblecore-api.yaml

```diff
@@ -3269,6 +3271,126 @@ paths:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/TestRunDto'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+    post:
+      tags:
+        - TestRun
+      summary: Create a test run
+      operationId: createTestRun
+      security:
+        - bearerAuth: []
+      requestBody:
+        description: 'Test run object that needs to be added to the database'
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/TestRunCreateInputDto'
+        required: true
+      responses:
+        '200':
+          description: Created test run
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TestRunDto'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '422':
+          $ref: '#/components/responses/422'
+  /scenario-testruns/{testRunId}:
+    get:
+      tags:
+        - TestRun
+      summary: Get a test run by id
+      operationId: getTestRun
+      security:
+        - bearerAuth: []
+      parameters:
+        - name: testRunId
+          description: ID of the test run that need to be fetched
+          in: path
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        '200':
+          description: The test run corresponding to the provided `testRunId`
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TestRunDto'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+    patch:
+      tags:
+        - TestRun
+      summary: Update a test run
+      operationId: updateTestRun
+      security:
+        - bearerAuth: []
+      parameters:
+        - name: testRunId
+          description: ID of the test run that need to be updated
+          in: path
+          required: true
+          schema:
+            type: string
+            format: uuid
+      requestBody:
+        description: 'Updated test run object'
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/TestRunUpdateInputDto'
+        required: true
+      responses:
+        '200':
+          description: The updated test run corresponding to the provided `testRunId`
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TestRunDto'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
```

Comment:
This is the endpoint I'm totally not fan of (any proposition is welcomed).

The ultimate goal is to allow to "down" an ongoing testrun.

Cancelling ongoing testrun looks necessary when you try to create a new one.
Still, this can be handled by the backend usecase, and we can remove this from the API. It won't prevent the frontend to display a warning modal (more or less like the information modal displayed on draft creation)

Reply:
let's decide about this later.. I think it is a product design issue; it is not necessary for the v0